### PR TITLE
Improve pattern_matching function documentation

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**

This PR improves pattern_matching function documentation.

### CODE REVIEW

File: pennylane/transforms/optimization/pattern_matching.py [hunk 1]
```diff
@@ -273,6 +275,56 @@ def pattern_matching(circuit_dag, pattern_dag):
...
+            qml.PauliZ(wires=0)
+            qml.S(wires=1)
+            qml.CZ(wires=[0, 1])
+            qml.S(wires=1)
+            qml.S(wires=2)
+            qml.CZ(wires=[1, 2])
+            qml.S(wires=2)
+            return qml.expval(qml.PauliX(wires=0))
+
+    where we want to find all maximal matches of a pattern containing a sequence of two ``pennylane.S`` gates and
```
Comment (antalszava) [inline pennylane/transforms/optimization/pattern_matching.py:296]:
```suggestion
    Assume that we want to find all maximal matches of a pattern containing a sequence of two :class:`~.S` gates and
```

File: pennylane/transforms/optimization/pattern_matching.py [hunk 2]
```diff
@@ -273,6 +275,56 @@ def pattern_matching(circuit_dag, pattern_dag):
...
+            qml.S(wires=0)
+            qml.S(wires=0)
+            qml.PauliZ(wires=0)
+
+    >>> circuit_dag = qml.commutation_dag(circuit)()
+    >>> pattern_dag = qml.commutation_dag(pattern)()
+    >>> all_max_matches = qml.pattern_matching(circuit_dag, pattern_dag)
+
+    It is possible to access the matches by looping through the list. The first integers indices represent the gates
+    in the pattern and the second intergers the gates in the circuit (by order of appearance).
```
Comment (antalszava) [inline pennylane/transforms/optimization/pattern_matching.py:312]:
```suggestion
    The matches are accessible by looping through the list outputted by `qml.pattern_matching`. This output is a list of two lists containing indices. The first list indexes the gates
    of the pattern and the second list provides indices for the gates in the circuit (by order of appearance).
```

Comment (antalszava) [reply pennylane/transforms/optimization/pattern_matching.py:312]: Also, a specific example explaining the indices and the gates in the pattern and the circuit would be great for the readers.

File: pennylane/transforms/optimization/pattern_matching.py [hunk 3]
```diff
@@ -273,6 +275,56 @@ def pattern_matching(circuit_dag, pattern_dag):
...
+
+    >>> for match_conf in all_max_matches:
+    ...     print(match_conf.match)
+
```
Comment (antalszava) [inline pennylane/transforms/optimization/pattern_matching.py]: The rendering is off here.
```suggestion
```

File: pennylane/transforms/optimization/pattern_matching.py [hunk 4]
```diff
@@ -273,6 +275,56 @@ def pattern_matching(circuit_dag, pattern_dag):
 
     Returns:
         list(Match): the list of maximal matches.
+
```
Comment (antalszava) [inline pennylane/transforms/optimization/pattern_matching.py:278]: How about having a `seealso` link here to `pattern_matching_optimization`?

**Review by antalszava — CHANGES_REQUESTED** [review]
Hi @rmoyard thank you! :tada: Left a couple of suggestions for readability.

File: pennylane/transforms/optimization/pattern_matching.py (hunk 4 above)
Comment (rmoyard) [reply pennylane/transforms/optimization/pattern_matching.py:278]: @antalszava I will add that!

**Review by antalszava — APPROVED** [review]
Thank you :tada:

**Comment by antalszava** [comment]
Note: the changes of this PR have been directly incorporated in https://github.com/PennyLaneAI/pennylane/pull/2474. Closing this PR.

**Pull request closed without merging**